templates/typescript_gapic/src/$version/$service_client.ts.njk

```diff
@@ -0,0 +1,307 @@
+              this._descriptors.stream[methodName] ||
+              this._descriptors.longrunning[methodName]);
+    }
+  }
+
+  /**
+   * The DNS address for this API service.
+   */
+  static get servicePath() {
+    return '{{ service.hostname }}';
+  }
+
+  /**
+   * The port for this API service.
+   */
+  static get port() {
+    return {{ service.port }};
+  }
+
+  /**
+   * The scopes needed to make gRPC calls for every method defined
+   * in this service.
+   */
+  static get scopes() {
+{%- if (service.oauthScopes.length == 0) %}
+    return [];
+{%- else %}
+    return [
+{%- set scopesJoiner = joiner() %}
+{%- for scope in service.oauthScopes %}
+      {{- scopesJoiner() }}
+      '{{ scope }}'
+{%- endfor %}
+    ];
+{%- endif %}
+  }
+
+  /**
+   * Return the project ID used by this class.
+   * @param {function(Error, string)} callback - the callback to
+   *   be called with the current project Id.
+   */
+  getProjectId(): Promise<string>;
+  getProjectId(callback: Callback<string, undefined, undefined>): void;
+  getProjectId(callback?: Callback<string, undefined, undefined>):
+      Promise<string>|void {
+    if (callback) {
+      this.auth.getProjectId(callback);
+      return;
+    }
+    return this.auth.getProjectId();
+  }
+
+  // -------------------
+  // -- Service calls --
+  // -------------------
+
+{%- for method in service.simpleMethods %}
+  {{ method.name.toCamelCase() }}(
+      request: protos{{ method.inputInterface }},
+      options?: gax.CallOptions):
+      Promise<[
+        protos{{ method.outputInterface }},
+        protos{{ method.inputInterface }}|undefined, {}|undefined
+      ]>;
+  {{ method.name.toCamelCase() }}(
+      request: protos{{ method.inputInterface }},
+      options: gax.CallOptions,
+      callback: Callback<
+          protos{{ method.outputInterface }},
+          protos{{ method.inputInterface }}|undefined,
+          {}|undefined>): void;
+  {{ method.name.toCamelCase() }}(
+      request: protos{{ method.inputInterface }},
+      optionsOrCallback?: gax.CallOptions|Callback<
+          protos{{ method.outputInterface }},
+          protos{{ method.inputInterface }}|undefined, {}|undefined>,
+      callback?: Callback<
+          protos{{ method.outputInterface }},
+          protos{{ method.inputInterface }}|undefined,
+          {}|undefined>):
+      Promise<[
+        protos{{ method.outputInterface }},
+        protos{{ method.inputInterface }}|undefined, {}|undefined
+      ]>|void {
+{%- if (method.streaming == 'SERVER_STREAMING' or method.streaming == 'NON_STREAMING') %}
+    request = request || {};
+{%- endif %}
+    let options = optionsOrCallback;
+    if (typeof options === 'function' && callback === undefined) {
+      callback = options;
+      options = {};
+    }
+    options = options || {};
+    return this._innerApiCalls.{{ method.name.toCamelCase() }}(request, options, callback);
+  } 
+{%- endfor%}
+
+  test() {
```

Comment:
that doesn't seem right . 🙃

Reply:
Add extra space after "endfor" and keep test in place.

Reply:
nah - I'm saying why do we have a `test` function at all in here?